Intune-PowerShell/DeviceConfiguration/DeviceConfigurationADMX_Export.ps1

```diff
@@ -526,87 +527,32 @@ foreach ($DCP in $DCPs)
-				$jsonCode = $jsonCode.Replace("<!PLACEHOLDER!>", $jsonCodePLACEHOLDER).Replace("True", "true")
-				write-host "Exporting setting $($DefinitionValuedefinitionDisplayName) to folder $ExportPath\$($FolderName)\$FileName.json" -ForegroundColor Yellow
-				New-Item "$ExportPath\$($FolderName)\$FileName.json" -ItemType File -Force
-				$jsonCode | Set-Content -LiteralPath "$ExportPath\$($FolderName)\$FileName.json" -Force
-			}
-			else
-			{
-				$jsonCode = @"
-{
-   "enabled":$($GroupPolicyConfigurationsDefinitionValue.enabled.tostring().Replace("True", "true")),
-   "presentationValues":[  
-      {  
-         <!PLACEHOLDER!>,
-         "presentation@odata.bind":"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$($DefinitionValuedefinitionID)')/presentations('$($GroupPolicyDefinitionsPresentations.id)')"
-      }
-   ],
-   "definition@odata.bind":"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$($DefinitionValuedefinitionID)')"
-}
-"@
-				
-				$jsonCodePLACEHOLDER = $JSON_Output.Substring(1)
-				$jsonCodePLACEHOLDER = $jsonCodePLACEHOLDER.Replace("<!PLACEHOLDER!>", $_).Substring(0, $jsonCodePLACEHOLDER.Length - 1)
-				$jsonCode = $jsonCode.Replace("<!PLACEHOLDER!>", $jsonCodePLACEHOLDER)
-				write-host "Exporting setting $($DefinitionValuedefinitionDisplayName) to folder $ExportPath\$($FolderName)\$FileName.json" -ForegroundColor Yellow
-				New-Item "$ExportPath\$($FolderName)\$FileName.json" -ItemType File -Force
-				$jsonCode | Set-Content -LiteralPath "$ExportPath\$($FolderName)\$FileName.json" -Force
-			}
-		}
-		else
-		{
-			
-			$jsonCode = @"
-{
-   "enabled":$($GroupPolicyConfigurationsDefinitionValue.enabled.tostring().Replace("True", "true").Replace("False", "false")),
-    "definition@odata.bind":"https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$($DefinitionValuedefinitionID)')"
-}
-"@
-			write-host "Exporting setting $($DefinitionValuedefinitionDisplayName) to folder $ExportPath\$($FolderName)\$FileName.json" -ForegroundColor Yellow
-			New-Item "$ExportPath\$($FolderName)\$FileName.json" -ItemType File -Force
-			$jsonCode | Out-File "$ExportPath\$($FolderName)\$FileName.json" -Encoding ascii -Force
-		}
-		
+		$DefinitionValuePresentationValues = Get-GroupPolicyConfigurationsDefinitionValuesPresentationValues -GroupPolicyConfigurationID $DCP.id -GroupPolicyConfigurationsDefinitionValueID $GroupPolicyConfigurationsDefinitionValue.id
+		$OutDef = New-Object -TypeName PSCustomObject
+        $OutDef | Add-Member -MemberType NoteProperty -Name "definition@odata.bind" -Value "https://graph.microsoft.com/beta/deviceManagement/groupPolicyDefinitions('$definitionValuedefinitionID')"
+        $OutDef | Add-Member -MemberType NoteProperty -Name "enabled" -value $($GroupPolicyConfigurationsDefinitionValue.enabled.tostring().tolower())
+        if ($DefinitionValuePresentationValues.count -gt 1) {
```

Comment:
should we remove this  if ($DefinitionValuePresentationValues.count -gt 1)  ?

Reply:
HI Sandy,
This line checks to see if there is multiple values to add the Presentation section, as that section is only required if there are properties within the admx policy that need to be defined.

if there are no properties, this section is ignored, but it is required for *complex* policies, for example if there is a textbox and radio buttons.

Reply:
yeah that shouldnt be an issue at all. going to resolve.